app/loader.cpp

```diff
@@ -112,6 +112,10 @@ void Loader::zoomToProject( QgsQuickMapSettings *mapSettings )
     QgsRectangle layerExtent = mapSettings->mapSettings().layerExtentToOutputExtent( layer, layer->extent() );
     extent.combineExtentWith( layerExtent );
   }
+  if ( extent.isEmpty() )
+  {
+    extent.combineExtentWith( QgsRectangle( 10.0, 10.0, 20.0, 20.0 ) );
+  }
```

Comment:
I am wondering about this arbitrary QgsRectangle... for a project with a single point, wouldn't it be better to use extent.grow(delta) to add some extra space around the point? E.g. delta 1000 for projected CRS and 0.01 for geographic CRS (which should both add ~1km space).

Reply:
You are welcome. Yes I thought the same. I didn't know about grow method. I thought it would be nice to use layer->crs().bounds().toString(2); but bounds() of CRS looks empty. I tested with WGS84, GRS80. If layer->crs()->isGeographic () 0.01 can be used. For one point projects, point's coordinates can be defined. I will try.

Reply:
The temporary solution also works for empty projects. The only use case is it enables GPS position, so a user can collect data from the current position. Some user may want to use empty project templates.

Reply:
Sorry I forgot to mention the issue #553 .

Reply:
I tried grow method, it works fine. I replaced the code with this:
```
if(extent.isEmpty()){
      if(layers.at(0)->crs().isGeographic()){
        extent.grow(0.01);
      }else{
          extent.grow(1000.0);
      }
  }
```

Reply:
Cool - this could crash though with an empty project because of the access to the first layer which could be null :)
How about something like this
```
extent.grow( mProject->crs().isGeographic() ? 0.01 : 1000.0 );
```

Reply:
That's much better idea. I didn't consider input would open a project without layer since there is nothing to input. 
I tested an empty project with a layer which doesn't have any feature.